### PR TITLE
GHA setup for debugging flaky test

### PR DESCRIPTION
**Pull request by ncooke3**
Attempt to reproduce #8652

Note, I disabled as many workflows and tests as possible so that only FIRMessagingTokenManagerTest is getting tested.

cc: @maksymmalyhin @chliangGoogle 

Hopefully, this PR can be a good way to repro the flaky tests and verify a fix.

#no-changelog

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -651,6 +651,9 @@ case "$product-$platform-$method" in
     RunXcodebuild \
       -scheme $product \
       "${xcb_flags[@]}" \
+      -test-iterations 10000 \
+      -retry-tests-on-failure \
+      -test-repetition-relaunch-enabled YES \
```
Comment (ncooke3) [inline scripts/build.sh:656]: Tried to repro the flaky test with Xcode 13's new test iteration flags. In the context of this PR, `FIRMessagingTokenManagerTest` ran 10000 times and I wasn't able to see any crashes.

Comment (paulb777) [reply scripts/build.sh:656]: Could the issue be related to running multiple incarnations of the tests in parallel?

Comment (ncooke3) [reply scripts/build.sh]: Yeah, good point. I added back all of Messaging's tests and have only seen one flake so far. 4a11772 ran 10000 times of the Messaging spm tests

Comment (paulb777) [reply scripts/build.sh]: Maybe try running all the platforms at the same time

Comment (paulb777) [reply scripts/build.sh]: cc @chliangGoogle

Comment (ncooke3) [reply scripts/build.sh]: > Maybe try running all the platforms at the same time

How would this done? @paulb777

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenManagerTest.m [hunk 2]
```diff
@@ -70,7 +72,10 @@ - (void)setUp {
 }
 
 - (void)tearDown {
-  [_messaging.tokenManager stopAllTokenOperations];
+  [_messaging.tokenManager.tokenOperations addBarrierBlock:^{
```
Comment (maksymmalyhin) [inline FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenManagerTest.m:75]: I guess, [waitUntilAllOperationsAreFinished](https://developer.apple.com/documentation/foundation/nsoperationqueue/1407971-waituntilalloperationsarefinishe) method may do the trick here.

Comment (ncooke3) [reply FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenManagerTest.m:75]: I tried that one and that one waits indefinitely.

Comment (ncooke3) [reply FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenManagerTest.m:75]: `stopAllTokenOperations` and `stopAllTokenOperations` both worked locally. `stopAllTokenOperations` didn't work as i expected it to on CI though.

Comment (maksymmalyhin) [reply FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenManagerTest.m:75]: In the case `waitUntilAllOperationsAreFinished` creates a deadlock, you may use `addBarrierBlock` along with an expectation to postpone tearDown completion.

Comment (maksymmalyhin) [reply FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenManagerTest.m:75]: If I'm not mistaken, `addBarrierBlock` returns immediately, so you will need an expectation here e.g.:
```
XCTestExpectation *exp = [...];
[_messaging.tokenManager.tokenOperations addBarrierBlock:^{
  [exp fulfill];
}];

[self waitForExpectations: @[exp]];
```

**Comment by ncooke3** [comment]
The current state here is that I know why the test is flaking but am having issues finding a solid solution. Solving the problem at hand leads to another problem cropping up.

**Comment by ncooke3** [comment]
Latest commit has a good solution to the problem.

Ran each test 1000 times with no failures. See GHA [messaging / spm (iOS) (pull_request)](https://github.com/firebase/firebase-ios-sdk/pull/8669/checks?check_run_id=3678724539)

**Comment by ncooke3** [comment]
Will open a separate PR with just that fix and close this one.

**Pull request closed without merging**